godot_project/project_workspace/workspace_context/history/history.gd

```diff
@@ -90,14 +90,18 @@ func register_snapshotable(in_system: Object) -> void:
 	_snapshotable_systems.append(in_system)
 
 
-func create_snapshot(in_snapshot_name: String) -> Dictionary:
-	_monitor_redundant_snapshots(in_snapshot_name)
+func create_snapshot_not_push() -> Dictionary:
```

Comment:
nit: Name should be `create_snapshot_no_push` (without the t)
or maybe it could just be an optional argument `add_to_stack` that default to true in the `create_snapshot` function